### PR TITLE
ENH -- Replaced for-loops in :function:`rescale_layout` with numpy vectorized methods.

### PR DESCRIPTION
**Pull request by it176131**
modified:   networkx/drawing/layout.py
- Replaced for-loops in :function:`rescale_layout` with numpy vectorized methods.
- No new tests as output should not have changed and no new features were introduced.
- Tested specifically with :method:`TestLayout.test_rescale_layout_dict` in drawing/tests/test_layout.py.
- All other tests ran per contributor guide.

### CODE REVIEW

File: networkx/drawing/layout.py [hunk 1]
```diff
@@ -1248,15 +1248,14 @@ def rescale_layout(pos, scale=1):
...
     """
+    import numpy as np
+
     # Find max length over all dimensions
-    lim = 0  # max coordinate for all axes
-    for i in range(pos.shape[1]):
-        pos[:, i] -= pos[:, i].mean()
-        lim = max(abs(pos[:, i]).max(), lim)
+    pos -= pos.mean(axis=0)
+    lim = max(np.abs(pos).max(), 0)  # max coordinate for all axes
```
Comment (rossbar) [inline networkx/drawing/layout.py:1255]: Given the subsequent `lim > 0`, I don't think the outer `max` is even necessary:

```suggestion
    lim = np.abs(pos).max()  # max coordinate for all axes
```

**Review by rossbar — APPROVED** [review]
Nice catch @it176131 , +1 for properly vectorizing this. The input of the functions are already numpy arrays so there's no worry about expanding the dependency footprint here. LGTM!

File: networkx/drawing/layout.py (hunk 1 above)
Comment (it176131) [reply networkx/drawing/layout.py:1255]: Good point. Will remove 👍

**Review by dschult — APPROVED** [review]
Looks good!  Thank you!

**Pull request merged**